contracts/Vouching.sol

```diff
@@ -1,216 +1,599 @@
+    }
+  }
+
+  /**
+   * @dev Rejects an appeal on a challenge. Can only be called by the overseer.
+   */
+  function overrule(uint256 _challengeID) public onlyOverseer existingChallenge(_challengeID) {
+    Challenge storage challenge_ = challenges_[_challengeID];
+    Appeal storage appeal_ = challenge_.appeal;
+    require(challenge_.resolution == Resolution.PENDING, "Given challenge was already resolved");
+    require(appeal_.appealer != address(0), "Cannot overrule a not-appealed challenge");
+
+    challenge_.resolution = Resolution.OVERRULED;
+    emit Overruled(_challengeID, overseer_);
+
+    uint256 i;
+    uint256 profit;
+    uint256 _blocked;
+    uint256 voucherProfit;
+    Voucher storage voucher_;
+    Entry storage entry_ = entries_[challenge_.entryID];
+
+    if (challenge_.answer == Answer.ACCEPTED) {
+      profit = appeal_.amount;
+      entry_.totalVouched = entry_.totalVouched.sub(challenge_.amount).add(profit);
+      entry_.totalAvailable = entry_.totalAvailable.add(profit);
+
+      for(i = 0; i < entry_.vouchersAddresses.length; i++) {
+        voucher_ = entry_.vouchers[entry_.vouchersAddresses[i]];
+        _blocked = voucher_.blockedPerChallenge[_challengeID];
+        if (_blocked > uint256(0)) {
+          voucherProfit = appeal_.amount.mul(_blocked).div(challenge_.amount);
+          voucher_.vouched = voucher_.vouched.sub(_blocked).add(voucherProfit);
+          voucher_.available = voucher_.available.add(voucherProfit);
+          // no need to reset the voucher's blocked amount for the challenge
+        }
+      }
+
+      token_.safeTransfer(challenge_.challenger, challenge_.amount.mul(2));
+    }
+    else {
+      profit = challenge_.amount.add(appeal_.amount);
+      entry_.totalVouched = entry_.totalVouched.add(profit);
+      entry_.totalAvailable = entry_.totalAvailable.add(challenge_.amount).add(profit);
+
+      for(i = 0; i < entry_.vouchersAddresses.length; i++) {
+        voucher_ = entry_.vouchers[entry_.vouchersAddresses[i]];
+        _blocked = voucher_.blockedPerChallenge[_challengeID];
+        if (_blocked > uint256(0)) {
+          uint256 appealPropProfit = appeal_.amount.mul(_blocked).div(challenge_.amount);
+          voucherProfit = _blocked.add(appealPropProfit);
+          voucher_.vouched = voucher_.vouched.add(voucherProfit);
+          voucher_.available = voucher_.available.add(_blocked).add(voucherProfit);
+          // no need to reset the voucher's blocked amount for the challenge
+        }
+      }
+    }
+  }
 
+  /**
+   * @dev Confirms the result of a challenge if it has not been appealed and the challenge period has passed.
+   * Transfers tokens associated to the challenge as needed.
+   */
+  function confirm(uint256 _challengeID) public existingChallenge(_challengeID) {
+    Challenge storage challenge_ = challenges_[_challengeID];
+    require(challenge_.answer != Answer.PENDING, "Cannot confirm a non-answered challenge");
```

Comment:
A pending challenge should be confirmable if the answer period has expired.

Reply:
Actually, anyone is allowed to answer a challenge if the answer period has expired. Thus, anyone can confirm a challenge after answering it.